### PR TITLE
Adopt juju 3.1

### PR DESCRIPTION
**Pull request by dashmage**

### CODE REVIEW

File: .github/workflows/check.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+  func:
+    uses: canonical/bootstack-actions/.github/workflows/func.yaml@v2
+    needs: lint-unit
+    with:
+      command: make functional
+      juju-channel: "3.1/stable"
+      nested-containers: true
+      python-version: "3.10"
+      timeout-minutes: 120
```
Comment (rgildein) [inline .github/workflows/check.yaml:31]: not-blocking: I think 60 min will be enough here

**Review by rgildein — APPROVED** [review]
LGTM

File: .github/workflows/check.yaml (hunk 1 above)
Comment (dashmage) [reply .github/workflows/check.yaml]: Done!

**Review by rgildein — APPROVED** [review]

File: .github/workflows/check.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+name: Check workflow running linter, unit and functional tests
```
Comment (agileshaw) [inline .github/workflows/check.yaml:1]: We also need to make changes in `sonar.yaml` ([L#9](https://github.com/canonical/prometheus-hardware-exporter/blob/main/.github/workflows/sonar.yaml#L9) and[ L#17](https://github.com/canonical/prometheus-hardware-exporter/blob/main/.github/workflows/sonar.yaml#L17)) to track this new workflow name.

**Review by agileshaw — APPROVED** [review]
LGTM, thanks

**Pull request merged**